File: intern/1200.minimum-absolute-difference.js
/*
 * @lc app=leetcode id=1200 lang=javascript
 *
 * [1200] Minimum Absolute Difference
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
const minimumAbsDifference = function (arr) {
    arr.sort((a, b) => a - b);
    let minDiff = Number.POSITIVE_INFINITY;
    let result = [];
    for (let i = 1; i < arr.length; i++) {
        const diff = arr[i] - arr[i - 1];
        if (diff < minDiff) {
            result = [];
            result.push([arr[i - 1], arr[i]]);
            minDiff = diff;
        } else if (diff === minDiff) {
            result.push([arr[i - 1], arr[i]]);
        }
    }
    return result;
};
// @lc code=end

File: src/2026/1200.minimum-absolute-difference.cpp
/*
 * @lc app=leetcode id=1200 lang=cpp
 *
 * [1200] Minimum Absolute Difference
 */

#include <algorithm>
#include <vector>

// @lc code=start
class Solution {
 public:
  std::vector<std::vector<int>> minimumAbsDifference(std::vector<int>& arr) {
    std::ranges::sort(arr);
    std::vector<std::vector<int>> pairs;
    int min_diff = INT_MAX;
    for (int i = 1; i < arr.size(); i++) {
      const int diff = arr[i] - arr[i - 1];
      if (diff == min_diff) {
        pairs.push_back({arr[i - 1], arr[i]});
      } else if (diff < min_diff) {
        pairs = {{arr[i - 1], arr[i]}};
      }
      min_diff = std::min(min_diff, diff);
    }

    return pairs;
  }
};
// @lc code=end
